File: 9-PalindromeNumber/9-PalindromeNumber.java
// Last updated: 8/5/2025, 12:36:57 PM
class Solution {
    public boolean isPalindrome(int x) {
            int rev = 0;
            int temp = x;
        if(x==0)
            return true;
    
        while(x>0){
            rev = x%10 + rev*10;
            x/=10;
        }
        if(rev==temp)
            return true;
        else
            return false;
        
    }
}

File: 9-PalindromeNumber/9-PalindromeNumber.py
# Last updated: 8/5/2025, 12:37:06 PM
class Solution:
    def isPalindrome(self, x: int) -> bool:
        temp,rev= x,0
        while temp>0:
            rev=(rev*10)+(temp%10)
            temp//=10
        return rev == x